### PR TITLE
added the function that fetches all exercises from supabase

### PR DESCRIPTION
**Pull request by Kepmon**
Nothing fancy here - just moved the list of exercises from a store to supabase and added a function that fetches all those exercises in the create page.

### CODE REVIEW

File: src/views/CreateView.vue
Comment (aleksanderwalczuk) [inline src/views/CreateView.vue]:
```js
const response = await supabase.from('Exercises').select()

if (response.ok) {
  this.allExercises = response.data;
} else {
  // show toast with error msg
}
```

File: src/views/CreateView.vue
Comment (aleksanderwalczuk) [inline src/views/CreateView.vue]: this should be awaited so `mounted` should be async

File: src/views/CreateView.vue
Comment (aleksanderwalczuk) [inline src/views/CreateView.vue]: I'd call it just `exercises`, WDYT?

File: src/App.vue
Comment (aleksanderwalczuk) [inline src/App.vue]: This change is unintended, it should be removed from this PR (you can literally create a new commit where it's deleted.

File: src/App.vue
Comment (aleksanderwalczuk) [inline src/App.vue]: This change is unintended, it should be removed from this PR (you can literally create a new commit where it's deleted.

File: src/App.vue
Comment (Kepmon) [reply src/App.vue]: **Done**

I will deal with the rest of your suggestions tomorrow since the addition of the error toast will probably require some more time than a few minutes but I did this one today so I don't forget tomorrow.

File: src/App.vue
Comment (Kepmon) [reply src/App.vue]: Same here.

Also deleted the relevant changes in the TheNav component.

File: src/views/CreateView.vue
Comment (Kepmon) [reply src/views/CreateView.vue]: **Corrected**, but could you please explain me what's the difference between:
```js
const { data } = await supabase
        .from('Exercises')
        .select()
```

and

```js
const response = await supabase.from('Exercises').select()
```

Also, I couldn't use `response.ok` because this response object didn't have such a property. So I changed it for `response.error === null`. Maybe I could do just `!response.error` but I seem to no longer know when I can use this syntax.

File: src/views/CreateView.vue
Comment (Kepmon) [reply src/views/CreateView.vue]: I believe I corrected that, not sure though. Should I now presumed that not only the functions is supposed to be asynchronous but the action of calling it as well?

File: src/views/CreateView.vue
Comment (Kepmon) [reply src/views/CreateView.vue]: Previously I had been afraid of a potential "collision" with `<transition name="exercises>` so I choose this name to avoid that but I changed for just `exercises` and everythin seems to work so you're probably right 🙃

File: src/views/CreateView.vue
Comment (aleksanderwalczuk) [inline src/views/CreateView.vue]: This can be avoided with object destruct

File: src/views/CreateView.vue
Comment (aleksanderwalczuk) [inline src/views/CreateView.vue]: Use destruct in parameter.
```js
this.exercises,filter(({ exercise }) => exercise.name.toLowerCase().includes(value))
```

File: src/views/CreateView.vue
Comment (aleksanderwalczuk) [inline src/views/CreateView.vue]: Are you sure that response error should be exactly null? I wonder if `responses.ok` isn't a better way

**Review by aleksanderwalczuk — CHANGES_REQUESTED** [review]
Minor fixes and it's ready

File: src/views/CreateView.vue
Comment (Kepmon) [reply src/views/CreateView.vue]: **Corrected**

Not sure if I did it in the way you wanted me to though.

File: src/views/CreateView.vue
Comment (Kepmon) [reply src/views/CreateView.vue]: I didn't actually need to destruct it after destructing the whole response in `mounted`.

File: src/views/CreateView.vue
Comment (Kepmon) [reply src/views/CreateView.vue]: I actually explained it here:
> Also, I couldn't use response.ok because this response object didn't have such a property. So I changed it for response.error === null. Maybe I could do just !response.error but I seem to no longer know when I can use this syntax.

I tried this `response.ok` at first, as you suggested it, but  my function didn't work so I logged `response.ok` and found out it returns `undefined`. So, I then logged just `response` and got this:
response

File: src/views/CreateView.vue
Comment (aleksanderwalczuk) [inline src/views/CreateView.vue]: Use map instead

Comment (aleksanderwalczuk) [reply src/views/CreateView.vue]:
```js
data.map(({muscle, img, name}) => ({ muscle, img, name})

```

Comment (aleksanderwalczuk) [reply src/views/CreateView.vue]: This method returns an array, so you can assign it's result to exercises in the same line 😁

**Review by aleksanderwalczuk — CHANGES_REQUESTED** [review]
One more to go. Apply the fix before merge

**Review by aleksanderwalczuk — APPROVED** [review]

File: src/views/CreateView.vue
Comment (Kepmon) [reply src/views/CreateView.vue]: I couldn't apply this the way you told me to because `muscle`, `img`, and `name` are properties of `data[x].exercise`, not of the data itself.

However, I changed it the way I could, please, let me know if this is ok. I won't merge this branch until I get the feedback from you regarding this matter.

File: src/views/CreateView.vue
Comment (aleksanderwalczuk) [reply src/views/CreateView.vue]: So you can check `status > 199 && status < 300`

File: src/views/CreateView.vue
Comment (aleksanderwalczuk) [reply src/views/CreateView.vue]: for one level nested prop I'd still consider using destruct.
```js
data.map(({ exercise: { muscle, img, name }}) => ({ muscle, img, name});
```

**Pull request merged**
